src/components/input-field/input-field.tsx

```diff
@@ -1,28 +1,36 @@
-import { type FC } from "react";
 import {
-  type ButtonProps,
   Stack,
-  type SxProps,
   TextField,
+  type ButtonProps,
   type TextFieldProps,
-  Typography,
 } from "@mui/material";
-import { SendButton } from "~/components/send-button/index.js";
-import { PrimaryButton } from "~/components/primary-button/index.js";
+import { type FC } from "react";
 import { text } from "~/assets/text/index.js";
+import { SendButton } from "~/components/send-button/index.js";
 import { styles } from "./styles.js";
 
-type Props = Pick<ButtonProps, "onClick"> & Omit<TextFieldProps, "onClick">;
+type Props = Pick<ButtonProps, "onClick"> &
+  Omit<TextFieldProps, "onClick"> & {
+    validationError?: string;
+  };
```

Comment:
I would still keep the text field onClick and declare your own onSubmit. It is clearer when you use the component

Reply:
```ts
interface Props extends TextFieldProps {
  validationError?: string;
  onSubmit: MouseEventHandler<HTMLButtonElement>
}
```

Reply:
if you choose to go the `onSubmit` route, it would need to be defined within a `<form>` tag in the parent component, and you would need to add `type="submit"` to the button within TextField (no need to pass an onSubmit prop) somthing like this

```
const TextFieldButton = () =>
  <>
    <InputField onChange={onChangeProp}/>
    <Button type="submit" />
  </>

const Parent = () => 
  <form onSubmit={handleSubmit}>
    <TextFieldButton onChange={...} />
  </form>
```

Reply:
`onClick` makes me think of clicking the input field if i am using the component. no matter the implementation that lays behind, what i am doing is submitting the data, thus it makes sense to call the event `onSubmit`. then that `onSubmit` can also be passed to the button's `onClick`, no need for a form. the important thing is that on a higher level I know clearly which action I am performing